From 0afb1a7175b75cc99f9d427dce75a58deb0c0dd5 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Wed, 6 Oct 2021 14:58:33 +0100
Subject: [PATCH] collect2: Fix missing cleanups.

The code that checks to see if objects have LTO content via
simple-object was not releasing resources, fixed thus.

Signed-off-by: Iain Sandoe <iain@sandoe.co.uk>

gcc/ChangeLog:

	* collect2.c (is_lto_object_file): Release simple-object
	resources, close files.

(cherry picked from commit 43ae43f654749d291d871ca6ef7c96ea16580fad)
---
 gcc/collect2.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git gcc/collect2.c gcc/collect2.c
index 8260863c04e..b3862daeedd 100644
--- gcc/collect2.c
+++ gcc/collect2.c
@@ -2301,10 +2301,15 @@ is_lto_object_file (const char *prog_name)
 							LTO_SEGMENT_NAME,
 							&errmsg, &err);
   if (!inobj)
-    return false;
+    {
+      close (infd);
+      return false;
+    }
 
   errmsg = simple_object_find_sections (inobj, has_lto_section,
 					(void *) &found, &err);
+  simple_object_release_read (inobj);
+  close (infd);
   if (! errmsg && found)
     return true;
 
-- 
2.40.1

